File: public/creational_patterns/builder/ConcreteBuilder.java

public class $CONCRETE_BUILDER_CLASSNAME1$ implements $BUILDER_CLASSNAME$ {

    private $PRODUCT_CLASSNAME$ product;

    public $CONCRETE_BUILDER_CLASSNAME1$() {
        this.product = new $PRODUCT_CLASSNAME$();
    }

$BUILDER_METHODS_EXTENDED$

    public $PRODUCT_CLASSNAME$ getResult() {
        return this.product;
    }
}

File: public/others/builder_cpp/ConcreteBuilder.cpp
class $CONCRETE_BUILDER_CLASSNAME$ : public $BUILDER_CLASSNAME$
{
private:
    $PRODUCT_CLASSNAME$ *product;

public:
    $CONCRETE_BUILDER_CLASSNAME$()
    {
        this->product = new $PRODUCT_CLASSNAME$();
    }

$BUILDER_METHODS_EXTENDED$

    $PRODUCT_CLASSNAME$ *GetProduct()
    {
        $PRODUCT_CLASSNAME$ *result = this->product;
        return result;
    }
};